web/src/main/java/org/kaaproject/kaa/sandbox/web/services/SandboxServiceImpl.java

```diff
@@ -559,6 +491,111 @@ public void buildProjectData(String uuid, BuildOutputData outputData, String pro
+                          Project project, FileData sdkFileData) throws IOException, SandboxServiceException {
+
+        outStream.println("Successfully got SDK.");
+        File rootDir = createTempDirectory("demo-project");
+        try {
+            outStream.println("Processing project archive...");
+            String sourceArchiveFile = Environment.getServerHomeDir() + "/" + DEMO_PROJECTS_FOLDER + "/" + project.getSourceArchive();
+            String rootProjectDir = rootDir.getAbsolutePath();
+
+            executeCommand(outStream, new String[]{"tar", "-C", rootProjectDir, "-xzvf", sourceArchiveFile}, null);
+
+            File sdkFile = new File(rootProjectDir + "/" + project.getSdkLibDir() + "/" + sdkFileData.getFileName());
+            FileOutputStream fos = FileUtils.openOutputStream(sdkFile);
+            fos.write(sdkFileData.getFileData());
+            fos.flush();
+            fos.close();
+
+            ProjectDataKey dataKey = new ProjectDataKey(projectId, dataType);
+            if (dataType == ProjectDataType.SOURCE) {
+                buildSdkSource(outStream, rootDir, sourceArchiveFile, rootProjectDir, dataKey);
+            } else {
+                buildBinary(outStream, project, rootDir, dataKey);
+            }
+        } finally {
+            FileUtils.deleteDirectory(rootDir);
+        }
+    }
+
+    private void buildBinary(ClientMessageOutputStream outStream, Project project, File rootDir, ProjectDataKey dataKey)
+            throws SandboxServiceException, IOException {
+
+        outStream.println("Building binary file...");
+        File projectFolder = rootDir;
+        if (project.getProjectFolder() != null && !project.getProjectFolder().trim().isEmpty()) {
+            projectFolder = new File(rootDir, project.getProjectFolder());
+        }
+
+        buildBinaryAntGradle(outStream, project, projectFolder);
+        outStream.println("Build finished.");
+
+        File binaryFile = new File(rootDir, project.getDestBinaryFile());
+        byte[] binaryFileBytes = FileUtils.readFileToByteArray(binaryFile);
+        FileData binaryFileData = new FileData();
+
+        String binaryFileName = FilenameUtils.getName(binaryFile.getAbsolutePath());
+
+        binaryFileData.setFileName(binaryFileName);
+        binaryFileData.setFileData(binaryFileBytes);
+        if (project.getSdkLanguage() == SdkLanguage.JAVA) {
+            if (project.getPlatforms().contains(Platform.ANDROID)) {
+                binaryFileData.setContentType("application/vnd.android.package-archive");
+            } else {
+                binaryFileData.setContentType("application/java-archive");
+            }
+        }
+        cacheService.putProjectFile(dataKey, binaryFileData);
+    }
+
+    private void buildBinaryAntGradle(ClientMessageOutputStream outStream, Project project, File projectFolder) throws SandboxServiceException {
```

Comment:
I think this method need to be inline inside 'buildBinary'. Reasons:
1) Method name is not perfect
2) It is private and used only in 'buildBinary'
3) I don't see use case when it will be used in other places.